src/components/auth/login/SignUpScreen.js

```diff
@@ -156,22 +162,26 @@ const SignupScreen = ({ isSignup, screenProps, styles, handleLoginMethod, sdkIni
                 {`${buttonPrefix} with Facebook`}
               </LoginButton>
 
-              <LoginButton
-                style={[
-                  styles.buttonLayout,
-                  styles.buttonsMargin,
-                  {
-                    backgroundColor: mainTheme.colors.darkBlue,
-                  },
-                ]}
-                onPress={_mobile}
-                disabled={!sdkInitialized}
-                testID="login_with_auth0"
-                icon={MobileBtnIcon}
-                iconProps={{ viewBox: '0 0 14.001 26' }}
-              >
-                {`${buttonPrefix}${isSignup ? '' : ' with'} Passwordless`}
-              </LoginButton>
+              {isMobileNative || !recaptchaVisible ? (
```

Comment:
1. move this condition inside web implementation
2. add prop to pass button's JSX (could do that via children e.g.`<recaptcha props here><button props here>`)
3. native should render both button + captcha, web should use condition and visibility state var